malariagen_data/anoph/dipclust.py

```diff
@@ -0,0 +1,276 @@
+from typing import Optional, Tuple
+
+import allel  # type: ignore
+import numpy as np
+from numpydoc_decorator import doc  # type: ignore
+
+from ..util import (
+    CacheMiss,
+    check_types,
+    multiallelic_diplotype_pdist,
+    multiallelic_diplotype_mean_sqeuclidean,
+    multiallelic_diplotype_mean_cityblock,
+)
+from ..plotly_dendrogram import plot_dendrogram
+from . import base_params, plotly_params, tree_params, dipclust_params
+from .base_params import DEFAULT
+from .snp_data import AnophelesSnpData
+
+
+class AnophelesDipClust(
```

Comment:
```suggestion
class AnophelesDipClustAnalysis(
```

Nit, could we add "Analysis" to the class name, just for naming consistency.